docs/sessions.md

```diff
@@ -264,6 +264,40 @@ sess.sql("SELECT * FROM example.tbl, temp LIMIT 1").show()
 
     We aim to support SQL DDL in future releases!
 
+### Using providers
+
+Various AI/ML protocols can be implemented through Providers. Here is an example of using OpenAI provider for creating text embeddings
+
+```python
+# Set the current provider to be OpenAI (requires a valid OpenAI API key)
+api_key = <your OpenAI API key here>
+sess.set_provider("openai", api_key=api_key)
```

Comment:
Perhaps include a note that the OPENAI_API_KEY environment variable will be picked up as well.